File: insertSort.js
let unsorted = [14, -4, 17, 6, 22, 1, -5];

function insertSort(arr) {
  //因為是假設arr的第一個index是已經排好的arry 所以for loop是從1開始
  for (let j = 1; j <= arr.length - 1; j++) {
    let key = arr[j];
    let i = j - 1;
    while (i >= 0 && arr[i] > key) {
      arr[i + 1] = arr[i];
      i--;
    }
    arr[i + 1] = key;
  }
  console.log(arr);
  return arr;
}

insertSort(unsorted);

File: insertSort.py
arr = [15,669,45,51,848,6,21,48,954,877,32,54,65,3345,31,248,35,8,9999]

def insertSort(arr):
    for i in range(1,len(arr),1):
        while (i > 0 and arr[i] < arr[i-1]):
            arr[i],arr[i-1] = arr[i-1], arr[i]
            i -= 1

if __name__ == '__main__':
    insertSort(arr)
    print(arr)